src/vmm/src/guest_config/aarch64_templates/mod.rs

```diff
@@ -0,0 +1,117 @@
+    #[error("Error getting register {0}: {1}")]
+    GetReg(u64, kvm_ioctls::Error),
+    /// Error setting register
+    #[error("Error setting register {0}: {1}")]
+    SetReg(u64, kvm_ioctls::Error),
+}
+
+/// Structure that represents change in the register
+/// We zero bits with zero mask and set with values mask
+/// (register & mask) | value
+pub struct RegisterValue {
+    mask: u128,
+    value: u128,
+}
+
+/// Cpu template for aarch64
+pub struct CpuTemplateAarch64(BTreeMap<u64, RegisterValue>);
+
+impl Deref for CpuTemplateAarch64 {
+    type Target = BTreeMap<u64, RegisterValue>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl CpuTemplateAarch64 {
+    /// Returns ids of registers that are changed
+    /// by this template
+    pub fn register_ids(&self) -> Vec<u64> {
+        self.keys().copied().collect()
+    }
+
+    /// Applies cpu template to vcpu
+    pub fn apply(&self, vcpu: &VcpuFd) -> std::result::Result<(), Error> {
+        for (reg_id, reg_value) in self.iter() {
+            let value = {
+                let curr_value = vcpu
+                    .get_one_reg(*reg_id)
+                    .map_err(|error| Error::GetReg(*reg_id, error))?;
+                (curr_value & reg_value.mask) | reg_value.value
+            };
+            vcpu.set_one_reg(*reg_id, value)
+                .map_err(|error| Error::SetReg(*reg_id, error))?;
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use arch::aarch64::regs::MPIDR_EL1;
+    use vm_memory::GuestMemoryMmap;
+
+    use super::*;
+    use crate::vcpu::KvmVcpu;
+    use crate::vstate::vm::tests::setup_vm;
+    use crate::vstate::vm::Vm;
+
+    fn setup_vcpu(mem_size: usize) -> (Vm, KvmVcpu, GuestMemoryMmap) {
+        let (mut vm, vm_mem) = setup_vm(mem_size);
+        let vcpu = KvmVcpu::new(0, &vm).unwrap();
+        vcpu.init(vm.fd()).unwrap();
+        vm.setup_irqchip(1).unwrap();
+
+        (vm, vcpu, vm_mem)
+    }
+
+    fn aarch64_cpu_template() -> CpuTemplateAarch64 {
+        CpuTemplateAarch64(
+            vec![(
+                MPIDR_EL1,
+                // Pretending that cpu is a part of multiprocessor system
+                RegisterValue {
+                    mask: 0b10111111111111111111111111111111,
+                    value: 0b00000000000000000000000000000000,
+                },
+            )]
+            .into_iter()
+            .collect(),
+        )
+    }
+
+    #[test]
+    fn test_register_ids() {
+        let template = aarch64_cpu_template();
+        let ids = template.register_ids();
+        assert_eq!(ids, vec![MPIDR_EL1]);
+    }
+
+    #[test]
+    fn test_apply_template() {
+        let (_, vcpu, _) = setup_vcpu(0x10000);
+        let template = aarch64_cpu_template();
+
+        assert!(template.apply(&vcpu.fd).is_ok());
+
+        let new_reg_value = vcpu.fd.get_one_reg(MPIDR_EL1).unwrap();
+        // checking that 30th bit is set to 0
+        assert_eq!(new_reg_value & (1 << 30), 0)
```

Comment:
How about checking all possible patterns `{(0, 0), (0, 1), (1, 0), (1, 1)}`?
```
                RegisterValue {
                    mask: 0b10011111111111111111111111111111,
                    value: 0b00110000000000000000000000000000,
                },
```

Reply:
what do you mean by all patterns? I change 1 bit and check that it is set to the value I need.

Reply:
Sorry, I was wrong. Please forget this.